### PR TITLE
[SPARK-6773][Tests]Fix RAT checks still passed issue when download rat jar failed

### PR DESCRIPTION
**Pull request by sisihj**
check -license will passed in next time when rat jar download failed.

Add 2 step to fix this:
1. Clean the rat.jar if download failed.
2. Add a check logic after run rat checking.

### CODE REVIEW

File: dev/check-license [hunk 1]
```diff
@@ -71,6 +72,11 @@ mkdir -p "$FWDIR"/lib
 
 $java_cmd -jar "$rat_jar" -E "$FWDIR"/.rat-excludes  -d "$FWDIR" > rat-results.txt
 
+if [ $? -ne 0 ]; then
+   echo "Error happen when running rat, pls checking the rat jar"
```
Comment (srowen) [inline dev/check-license]: This seems fine except that this message needs improvement. 'RAT exited abnormally' is probably fine.

**Comment by sisihj** [comment]
@srowen  fixed according you comments, pls review.

File: dev/check-license [hunk 2]
```diff
@@ -43,6 +43,7 @@ acquire_rat_jar () {
     unzip -tq $JAR &> /dev/null
     if [ $? -ne 0 ]; then
       # We failed to download
+      rm $JAR
```
Comment (srowen) [inline dev/check-license]: Actually I also just noticed that this should be quoted as `"$JAR"`, and above in the `unzip` command, to take care with paths containing spaces.

**Comment by srowen** [comment]
Looking again at the rest of the script, I don't see the point of `if [[ ! -f "$rat_jar" ]]; then`, since it's redundant with the check that follows. Both enclose the same code (excepting that final print statement, which doesn't matter). @ScrapCodes is that right? https://github.com/apache/spark/commit/21109fbab0ded4f73cf4f8816859c76f125adfea

**Comment by ScrapCodes** [comment]
You are right, that check is just redundant ! + This PR LGTM given the fix that @srowen suggested is done.

**Comment by srowen** [comment]
ok to test

**Pull request closed without merging**